wp/lib/bap_wp/src/precondition.ml

```diff
@@ -833,6 +796,7 @@ let check ?refute:(refute = true) (solver : Solver.solver) (ctx : Z3.context)
     else
       pre'
   in
+  Printf.printf "Z3 Query:\n%s\n" (Z3.Expr.to_string (Z3.Expr.simplify is_correct None));
```

Comment:
Do you think we can make this an `info` log or something so that it doesn't print out when we run the unit tests?

Reply:
Seems reasonable. Or perhaps a command line flag.
Is there an insane amount of stuff in info? I would be of the opinion we also suppress the Constrain being printed by default and maybe the BIL.

Reply:
I'm not against having a flag for `verbose` mode, but it's probably out of the scope for this PR. And there isn't too much stuff in info. From grepping, I can see we have 8 info logs.

Reply:
Yea `info` seems reasonable... IDK.